its/src/test/java/com/sonar/it/scanner/msbuild/ScannerMSBuildTest.java

```diff
@@ -937,28 +951,73 @@ public void incrementalPrAnalysis_NoCache() throws IOException {
+    // - archive the content of `.sonarqube\out\.sonar\scanner-report\` (only the content, without parent folder) as `scanner-report.zip`
+    File reportZip = projectDir.resolve("scanner-report.zip").toFile();
+
+    uploadAnalysisWithCache(projectKey, baseBranch, reportZip);
+
+    File fileToBeChanged = projectDir.resolve("IncrementalPRAnalysis\\WithChanges.cs").toFile();
+    BufferedWriter writer = new BufferedWriter(new FileWriter(fileToBeChanged, true));
+    writer.append(' ');
+    writer.close();
+
     BuildResult result = ORCHESTRATOR.executeBuild(TestUtils.newScanner(ORCHESTRATOR, projectDir)
       .addArgument("begin")
       .setProjectKey(projectKey)
       .setProperty("sonar.projectBaseDir", projectDir.toAbsolutePath().toString())
       .setDebugLogs(true) // To assert debug logs too
-      .setProperty("sonar.pullrequest.base", "base-branch"));
+      .setProperty("sonar.pullrequest.base", baseBranch));
 
     assertTrue(result.isSuccess());
     assertThat(result.getLogs()).contains("Processing analysis cache");
-    assertThat(result.getLogs()).contains("Processing pull request with base branch 'base-branch'.");
-    assertThat(result.getLogs()).contains("Downloading cache. Project key: incremental-pr-analysis, branch: base-branch.");
-    // ToDo: Uncomment these assertions
-    //    assertThat(expectedUnchangedFiles).exists();
-    //    assertThat(Files.readString(expectedUnchangedFiles.toPath()))
-    //      .contains("Unchanged1.cs")
-    //      .contains("Unchanged2.cs")
-    //      .doesNotContain("WithChanges.cs"); // Because it was modified
+    assertThat(result.getLogs()).contains("Processing pull request with base branch 'master'.");
+    assertThat(result.getLogs()).contains("Downloading cache. Project key: incremental-pr-analysis, branch: master.");
+
+    File expectedUnchangedFiles = new File(projectDir.resolve(".sonarqube\\conf\\UnchangedFiles.txt").toString());
+
+    assertThat(expectedUnchangedFiles).exists();
+        assertThat(Files.readString(expectedUnchangedFiles.toPath()))
+          .contains("Unchanged1.cs")
+          .contains("Unchanged2.cs")
+          .doesNotContain("WithChanges.cs"); // Was modified
+  }
+
+  private void uploadAnalysisWithCache(String projectKey, String baseBranch, File reportZip) throws IOException {
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+    HttpPost uploadFile = new HttpPost(ORCHESTRATOR.getServer().getUrl() + "/api/ce/submit");
+
+    MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+    builder.addTextBody("projectKey", projectKey);
+    builder.addBinaryBody("report", new FileInputStream(reportZip), ContentType.APPLICATION_OCTET_STREAM, reportZip.getName());
+
+    HttpEntity multipart = builder.build();
+    uploadFile.setEntity(multipart);
+    httpClient.execute(uploadFile);
+
+    await()
+      .pollInterval(Duration.ofSeconds(1))
```

Comment:
Just to be explicit: 1s is the pooling interval (it will try once a second) and the maximum waiting time is 20 seconds.

Locally it's very fast. It works in 1 second but it looks like on IT's, it's not fast enough. I will try to increase the number.